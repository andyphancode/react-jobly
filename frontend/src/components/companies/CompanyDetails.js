import React, { useState, useEffect } from 'react';
import { useParams } from "react-router-dom";
import JoblyApi from '../../api/api';
import LoadingSpinner from '../../LoadingSpinner';
import JobCardList from '../jobs/JobCardList';

/**
 * 
 *  Company detail component rendering information about company. Also lists jobs. Accessible at /companies/:handle
 * 
 */

function CompanyDetails() {
  
  const { handle } = useParams();
  const [company, setCompany] = useState(null);
  
  useEffect(function getCompanyDetails() {
    async function getCompany() {
      setCompany(await JoblyApi.getCompany(handle));
    }

    getCompany();
  }, [handle])

  if (!company) return <LoadingSpinner />;

  return (
    <div className="CompanyDetails">
      <h5>{company.name}</h5>
      <p>{company.description}</p>
      <JobCardList jobs={company.jobs}/>
    </div>
  );
};

export default CompanyDetails;