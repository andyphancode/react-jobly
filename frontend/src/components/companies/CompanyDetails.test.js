import React from "react";
import { render } from "@testing-library/react";
import CompanyDetails from "./CompanyDetails";
import { MemoryRouter, Routes, Route } from "react-router-dom";

it("renders without crashing", function () {
  render(
      <MemoryRouter>
        <CompanyDetails />
      </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter initialEntries={["/companies/moore-plc"]}>
        <Routes>
            <Route path="/companies/:handle" element={<CompanyDetails/>} />            
        </Routes>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

