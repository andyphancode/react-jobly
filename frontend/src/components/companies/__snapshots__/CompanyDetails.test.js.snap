// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="LoadingSpinner"
  >
    <h1
      aria-label="hourglass"
      class="spinning-text"
      role="img"
    >
      âŒ›
    </h1>
    Fetching results...
  </div>
</DocumentFragment>
`;
